WooCommerce/Classes/ViewRelated/Products/Variations/Edit Attributes/RenameAttributesViewController.swift

```diff
@@ -0,0 +1,228 @@
+        }
+        presentBackNavigationActionSheet()
+        return false
+    }
+
+    override func shouldPopOnSwipeBack() -> Bool {
+        return shouldPopOnBackButton()
+    }
+
+    private func presentBackNavigationActionSheet() {
+        UIAlertController.presentDiscardChangesActionSheet(viewController: self, onDiscard: { [weak self] in
+            self?.navigationController?.popViewController(animated: true)
+        })
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension RenameAttributesViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = sections[indexPath.section].rows[indexPath.row]
+
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension RenameAttributesViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        return sections[section].footer
+    }
+}
+
+// MARK: - Support for UITableViewDataSource
+//
+private extension RenameAttributesViewController {
+   func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as TextFieldTableViewCell:
+            configureAttributeName(cell: cell)
+        default:
+            fatalError("Unidentified attribute row type")
+        }
+    }
+
+    func configureAttributeName(cell: TextFieldTableViewCell) {
+        cell.accessoryType = .none
+
+        let placeholder = Localization.placeholder
+
+        let cellViewModel = TextFieldTableViewCell.ViewModel(text: viewModel.attributeName,
+                                                         placeholder: placeholder,
+                                                         onTextChange: { [weak self] newAttributeName in
+                                                            self?.viewModel.handleAttributeNameChange(newAttributeName)
+                                                            self?.enableDoneButton(self?.viewModel.newAttributeName != nil)
```

Comment:
I think it's worth to move this logic inside the `ViewModel` and provide an output like `shouldEnableDoneBotton`, that way we can add a test for it!

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/3768/commits/bb36f3908b1c20488ea4fe04366b9785a2658d5b! I realized I'd forgotten to add unit tests in this PR so I added those in https://github.com/woocommerce/woocommerce-ios/pull/3768/commits/6b3eb8dcf406268439ebf104aae8a3f76323e457. Thanks for the nudge!